source/Color.cpp

```diff
@@ -74,7 +78,29 @@ bool Color::IsLoaded() const
 // Get a float vector representing this color, for use by OpenGL.
 const float *Color::Get() const
 {
-	return color;
+	static float c[4];
+	std::copy(std::begin(color), std::end(color), std::begin(c));
+
+	switch(Preferences::GetColorFilterMode())
+	{
+		case Preferences::ColorFilter::NORMAL:
+			break;
+		// Color blindness accessibility filters are enabled.
+		case Preferences::ColorFilter::PROTANOPIA:
+			if(color[0] > color[1])
+				c[2] = (1 - (1 - color[2]) * (1 - color[0]) + color[2]) / 2;
+			break;
+		case Preferences::ColorFilter::DEUTERANOPIA:
+			if(color[1] > color[0])
+				c[2] = (1 - (1 - color[2]) * (1 - color[1]) + color[2]) / 2;
+			break;
+		case Preferences::ColorFilter::TRITANOPIA:
+			c[0] += (color[1] - color[0]) * color[2];
+			c[1] += (color[0] - color[1]) * color[2];
+			break;
```

Comment:
The PR description says that you ran these filters through a color blindness simulator and checked the results, but why were these exact modifications chosen? Could perhaps have a comment explaining where you got these from.

Reply:
I've added some more detailed comments, but I'll keep this unresolved in case they weren't what you were looking for.